e2etest/l2tests/assignment.go

```diff
@@ -515,4 +518,57 @@ var _ = ginkgo.Describe("IP Assignment", func() {
 			framework.ExpectNoError(err)
 		})
 	})
+
+	ginkgo.Context("IPV4 - assign specifc IP for LB service", func() {
+		ginkgo.It("should preseve the same external ip after controller restart", func() {
+			specificIP := "192.168.10.100"
+
+			resources := config.Resources{
+				Pools: []metallbv1beta1.IPAddressPool{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "assignment-controller-reset-test-pool",
+						},
+						Spec: metallbv1beta1.IPAddressPoolSpec{
+							Addresses: []string{"192.168.10.100/32"},
+						},
+					},
+				},
+			}
+			err := ConfigUpdater.Update(resources)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("creating a service LB with a specific IP and two other LB services")
+			jig := e2eservice.NewTestJig(cs, f.Namespace.Name, "service-specific")
+			specificService, err := jig.CreateLoadBalancerService(30*time.Second, func(svc *v1.Service) { service.WithSpecificIPs(svc, specificIP) })
+			framework.ExpectNoError(err)
+			defer service.Delete(cs, specificService)
+			service.ValidateDesiredLB(specificService)
+
+			svc, err := jig.CreateLoadBalancerServiceWaitForClusterIPOnly(func(svc *v1.Service) { svc.ObjectMeta.Name = "service-b" })
+			framework.ExpectNoError(err)
+			defer service.Delete(cs, svc)
+
+			svc, err = jig.CreateLoadBalancerServiceWaitForClusterIPOnly(func(svc *v1.Service) { svc.ObjectMeta.Name = "service-c" })
+			framework.ExpectNoError(err)
+			defer service.Delete(cs, svc)
+
+			ginkgo.By("validating metallb keeps the specific ip after controller restart")
+			gomega.Consistently(func() error {
+				metallb.RestartController(cs)
+
+				// sleep test to let the controller time to initialize work
+				time.Sleep(5 * time.Second)
+
+				specificService, err = cs.CoreV1().Services(specificService.Namespace).Get(context.TODO(), specificService.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+
+				if len(specificService.Status.LoadBalancer.Ingress) == 0 {
+					return fmt.Errorf("ip assigned to the LB service %s was removed", specificService.Name)
+				}
+
+				return nil
+			}, 2*time.Minute, 1*time.Second).ShouldNot(gomega.HaveOccurred())
```

Comment:
1 second here means we'll try to restart the controller (and wait 5 secs) every second. I'd relax it to 30 seconds / 1 minute.